packages/astro/src/assets/services/service.ts

```diff
@@ -140,88 +141,88 @@ export type BaseServiceTransform = {
-				message: AstroErrorData.ExpectedImage.message(
-					JSON.stringify(options.src),
-					typeof options.src,
-					JSON.stringify(options, (_, v) => (v === undefined ? null : v))
-				),
-			});
-		}
-
-		if (!isESMImportedImage(options.src)) {
-			// User passed an `/@fs/` path or a filesystem path instead of the full image.
-			if (
-				options.src.startsWith('/@fs/') ||
-				(!isRemotePath(options.src) && !options.src.startsWith('/'))
-			) {
-				throw new AstroError({
-					...AstroErrorData.LocalImageUsedWrongly,
-					message: AstroErrorData.LocalImageUsedWrongly.message(options.src),
-				});
-			}
-
-			// For remote images, width and height are explicitly required as we can't infer them from the file
-			let missingDimension: 'width' | 'height' | 'both' | undefined;
-			if (!options.width && !options.height) {
-				missingDimension = 'both';
-			} else if (!options.width && options.height) {
-				missingDimension = 'width';
-			} else if (options.width && !options.height) {
-				missingDimension = 'height';
-			}
-
-			if (missingDimension) {
-				throw new AstroError({
-					...AstroErrorData.MissingImageDimension,
-					message: AstroErrorData.MissingImageDimension.message(missingDimension, options.src),
-				});
-			}
-		} else {
-			if (!VALID_SUPPORTED_FORMATS.includes(options.src.format as any)) {
-				throw new AstroError({
-					...AstroErrorData.UnsupportedImageFormat,
-					message: AstroErrorData.UnsupportedImageFormat.message(
-						options.src.format,
-						options.src.src,
-						VALID_SUPPORTED_FORMATS
-					),
-				});
-			}
-
-			if (options.widths && options.densities) {
-				throw new AstroError(AstroErrorData.IncompatibleDescriptorOptions);
-			}
-
-			// We currently do not support processing SVGs, so whenever the input format is a SVG, force the output to also be one
-			if (options.src.format === 'svg') {
-				options.format = 'svg';
-			}
-
-			if (
-				(options.src.format === 'svg' && options.format !== 'svg') ||
-				(options.src.format !== 'svg' && options.format === 'svg')
-			) {
-				throw new AstroError(AstroErrorData.UnsupportedImageConversion);
+	async validateOptions(options) {
+        // `src` is missing or is `undefined`.
+        if (!options.src || (typeof options.src !== 'string' && typeof options.src !== 'object')) {
+            throw new AstroError({
+                ...AstroErrorData.ExpectedImage,
+                message: AstroErrorData.ExpectedImage.message(
+                    JSON.stringify(options.src),
+                    typeof options.src,
+                    JSON.stringify(options)
+                ),
+            });
+        }
+
+        // Infer size for remote images if inferSize is true
+        if (options.inferSize && !isESMImportedImage(options.src)) {
+					try {
+							const result = await probe(options.src); // Directly probe the image URL
+							options.width = options.width || result.width;
+							options.height = options.height || result.height;
```

Comment:
```suggestion
							options.width ??= result.width;
							options.height ??= result.height;
```
seems like that would work too?